search_params.go

```diff
@@ -75,7 +75,7 @@ func NewSearchParams(idx Index, params json.RawMessage, sel *C.FaissIDSelector,
 		if ivfParams.NprobePct > 0 {
```

Comment:
To confirm with team: In case of a user specified nprobePct, should the percentage be applied on _all_ the clusters in the index, or only those with at least 1 filter hit?
Eg. If only 700 clusters out of a 1000 cluster IVF index have at least 1 filtered hit, should the user-specified percentage of 1% be applied on the 700 clusters(top 7) or the 1000(top 10)?

Reply:
@abhinavdangeti what's your take?

Reply:
I concur. This PR has the changes required to limit nProbePct to only clusters with eligible hits.